### PR TITLE
Fix tweak encoding

### PR DESCRIPTION
**Pull request by m-cat**
Fixes tweaks being erroneously encoded to utf-8 strings. Because the tweaks contain random bytes they are not valid Unicode byte sequences and this was causing much of the resulting strings to contain repeated "not a character" code points. Also, the encoding was different between browser and Node.

Instead of encoding tweaks to UTF-8 we instead hex-encode them and pass them directly to SkyDB with the new `hashedDataKeyHex` option. This lets SkyDB know that it is already a hash so that it does not hash the data key again.

### CODE REVIEW

**Review by redsolver — DISMISSED** [review]

File: src/crypto.ts [hunk 1]
```diff
@@ -166,12 +166,15 @@ export function hashDataKey(datakey: string): Uint8Array {
  * @param registryEntry - Registry entry to hash.
  * @returns - Hash of the registry entry.
  */
-export function hashRegistryEntry(registryEntry: RegistryEntry): Uint8Array {
-  return hashAll(
-    hashDataKey(registryEntry.datakey),
-    encodeString(registryEntry.data),
-    encodeBigintAsUint64(registryEntry.revision)
-  );
+export function hashRegistryEntry(registryEntry: RegistryEntry, hashedDataKeyHex: boolean): Uint8Array {
```
Comment (peterjan) [inline src/crypto.ts:169]: missing `@param`, docstring could use a comment explaining what it does exactly or how it influences the behaviour of the function

File: src/utils/url.ts [hunk 2]
```diff
@@ -98,12 +98,15 @@ export function getEntryUrlForPortal(
     throw new Error(`Given public key '${publicKey}' is not a valid hex-encoded string or contains an invalid prefix`);
   }
 
-  // We need to hash the data key in order to form the correct URL, as Sia hashes whatever data key we give it.
-  const dataKeyHash = toHexString(hashDataKey(dataKey));
+  // If not a hashed byte array already, hash the data key in order to form the correct URL.
```
Comment (peterjan) [inline src/utils/url.ts:101]: is this correct? Shouldn't it be `Hash and hex encode the given datakey if the given data key is not a hash already`?

File: src/integration.test.ts [hunk 3]
```diff
@@ -31,7 +31,38 @@ expect.extend({
...
+      const { data: received } = await client.file.getJSON(userID, path);
+      expect(received).toEqual(expect.objectContaining(expected));
+    });
+  });
+
   describe("SkyDB end to end integration tests", () => {
+    it("Should get existing SkyDB data", async () => {
+      const publicKey = "4a964fa1cb329d066aedcf7fc03a249eeea3cf2461811090b287daaaec37ab36";
+      const dataKey = "dataKey1";
+      const expected = { message: "hi Marcin" };
```
Comment (peterjan) [inline src/integration.test.ts:49]: lol

**Review by peterjan — APPROVED** [review]

**Review by peterjan — DISMISSED** [review]

File: src/crypto.test.ts [hunk 4]
```diff
@@ -161,11 +153,14 @@ describe("hashRegistryValue", () => {
 
     // "h" is the hash generated by siad with the same input parameters
     const h = "ff3b430675a0666e7461bc34aec9f66e21183d061f0b8232dd28ca90cc6ea5ca";
-    const hash = hashRegistryEntry({
-      datakey: "HelloWorld π",
-      data: "abc π",
-      revision: BigInt(123456789),
-    });
+    const hash = hashRegistryEntry(
```
Comment (peterjan) [inline src/crypto.test.ts:156]: Maybe add a test case where `hashDataKeyHex` is `true`, asserting the output is what we expect?
Or assert that 
```
hashRegistryEntry(
      {
        datakey: "HelloWorld",
        data: "abc",
        revision: BigInt(123456789),
      },
      false
    )
``` 
!= 
```
hashRegistryEntry(
      {
        datakey: "HelloWorld",
        data: "abc",
        revision: BigInt(123456789),
      },
      true
    )
```

File: src/utils/url.ts (hunk 2 above)
Comment (m-cat) [reply src/utils/url.ts:101]: Good catch, pertained to a previous commit

**Pull request merged**